src/x_connective.c

```diff
@@ -1365,10 +1370,12 @@ static void makefilename_float(t_makefilename *x, t_floatarg f)
         sprintf(buf, "%s",  x->x_format->s_name);
         break;
     case INT:
-        sprintf(buf, x->x_format->s_name, (int)f);
+        sprintf(buf, x->x_format->s_name,
+            x->x_long ? (long)f : (int)f);
```

Comment:
this may not work, as the ternary operator can only return a single type, so it promotes the `(int)f` to a `long` before passing it to `sprintf()`.

Reply:
An 'if' statement should solve this the, huh?

Reply:
> An 'if' statement should solve this then, huh?

@umlaeute did you see https://github.com/pure-data/pure-data/pull/2510/commits/81d6cdbb2f18a0f2837e04d3e8f0fbe6ac4df375 ? I tried using an if statement